gdal-spark/src/test/scala/geotrellis/spark/gdal/GDALRasterSourceRDDSpec.scala

```diff
@@ -0,0 +1,404 @@
+            List(IO {
+              // println(Thread.currentThread().getName())
+              // Thread.sleep((Math.random() * 100).toLong)
+              val lts = reprojRS(i)
+              lts.readAll(lts.keys.take(10).toIterator)
+              reprojRS(i).source.resolutions
+
+              dirtyCalls(reprojRS(i).source)
+            }, IO {
+              // println(Thread.currentThread().getName())
+              // Thread.sleep((Math.random() * 100).toLong)
+              val lts = reprojRS(i)
+              lts.readAll(lts.keys.take(10).toIterator)
+              reprojRS(i).source.resolutions
+
+              dirtyCalls(reprojRS(i).source)
+            }, IO {
+              // println(Thread.currentThread().getName())
+              // Thread.sleep((Math.random() * 100).toLong)
+              val lts = reprojRS(i)
+              lts.readAll(lts.keys.take(10).toIterator)
+              reprojRS(i).source.resolutions
+
+              dirtyCalls(reprojRS(i).source)
+            })
+          }
+        }.parSequence.unsafeRunSync
+
+        println(java.lang.Thread.activeCount())
+
+        res
+      }
+
+      it(s"should not fail on parallelization with a fork join pool") {
+        val i = 1000
+        implicit val cs = IO.contextShift(ExecutionContext.global)
+
+        val res = parellSpec(i)
+      }
+
+      it(s"should not fail on parallelization with a fixed thread pool") {
+        val i = 1000
+        val n = 200
+        val pool = Executors.newFixedThreadPool(n)
+        val ec = ExecutionContext.fromExecutor(pool)
+        implicit val cs = IO.contextShift(ec)
+
+        val res = parellSpec(i)
+      }
+    }
+  }
+
+  describe("RasterSourceRDD.read") {
+
+    val floatingScheme = FloatingLayoutScheme(500, 270)
+    val floatingLayout = floatingScheme.levelFor(rasterSource.extent, rasterSource.cellSize).layout
+
+    val cellType = rasterSource.cellType
+
+    val multibandTilePath = sparkGeoTiffPath("vlm/aspect-tiled-0-1-2.tif")
+
+    val noDataTile = ArrayTile.alloc(cellType, rasterSource.cols.toInt, rasterSource.rows.toInt).fill(NODATA).interpretAs(cellType)
+
+    val paths: Seq[String] = 0 to 5 map { index => filePathByIndex(index) }
+
+    lazy val expectedMultibandTile = {
+      val tiles = paths.map { MultibandGeoTiff(_, streaming = false).tile.band(0) }
+      MultibandTile(tiles)
+    }
+
+    it("should read in singleband tiles") {
+      val readingSources: Seq[ReadingSource] = paths.zipWithIndex.map { case (path, index) => ReadingSource(GeoTiffRasterSource(path), 0, index) }
+
+      val expected = expectedMultibandTile
+      val actual = RasterSourceRDD.read(readingSources, floatingLayout).stitch()
+
+      // random chip to test agains, to speed up tests
+      val gridBounds = RasterExtent(randomExtentWithin(actual.extent), actual.cellSize).gridBounds
+
+      expected.dimensions shouldBe actual.dimensions
+
+      assertEqual(expected.crop(gridBounds), actual.tile.crop(gridBounds))
```

Comment:
Full rasters comparison is too slow, since it is ~`1500x1500` n bands comparison, so I decided to generate a tiny (~250x250) random grid bounds to test it; it added an x10 boost to theese tests (less than 10 seconds instead of more than a minute)